onnxruntime/core/framework/data_transfer_manager.h

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/common/status.h"
+#include "core/framework/data_transfer.h"
+#include "core/framework/tensor.h"
+
+namespace onnxruntime {
+
+// Data transfer manager, which has all functions registered to copy tensors with different location.
+// It's not thread-safe.
+class DataTransferManager {
+ public:
+  DataTransferManager() = default;
+  //static DataTransferManager& Instance();
+
+  common::Status RegisterDataTransfer(std::unique_ptr<IDataTransfer> data_transfer);
+
+  common::Status CopyTensor(const Tensor& src, Tensor& dst) const;
+  common::Status CopyTensor(const Tensor& src, Tensor& dst, int exec_queue_id) const;
```

Comment:
Maybe CUDA can register two DataTransfer? One for copy in, and one for copy out. That way there's no need for exec_queue_id here.

Reply:
will do the refactor later.